tests/topotests/lib/evpn.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: ISC
+#
+# Copyright (c) 2025 by 6WIND
+# Used Copyright (c) 2018 by Network Device Education Foundation, Inc.
+# ("NetDEF") in this file.
+#
+
+from functools import partial
+from lib import topotest
+from lib.bgp import verify_bgp_rib
+
+from lib.topolog import logger
+
+
+def evpn_ping_router(pingrouter, ipv4_only=False, ipv6_only=False):
```

Comment:
it makes sense to move the functions into lib, but it should have more parameters like source_vrf and destination_ips to be usable in other tests. In addition it might be good to differentiate if the pingrouter is running in netns mode (as far as I can see pingrouter.net.use_netns_vrf should have that), otherwise using `ping -I <vrf name>`.

Reply:
done. I propose to replace ipv4_only and ipv6_only boolean with optional ipv4_address and ipv6_address. The presence of the parameter (not set to None) means that we want to use ping over ipv4 or ping over ipv6 or both.